sequencer/batchbuilder.go

```diff
@@ -243,23 +243,61 @@ func (s *Sequencer) closeSequence(ctx context.Context) error {
 // in case it's enough blocks since last GER update, long time since last batch and sequence is profitable
 func (s *Sequencer) shouldCloseSequenceInProgress(ctx context.Context) bool {
 	// Check if GER needs to be updated
-	numberOfBlocks, err := s.state.GetNumberOfBlocksSinceLastGERUpdate(ctx, nil)
-	if err != nil && err != state.ErrNotFound {
-		log.Errorf("failed to get last time GER updated, err: %v", err)
+	blockNum, mainnetExitRoot, err := s.state.GetBlockNumAndMainnetExitRootByGER(ctx, s.sequenceInProgress.GlobalExitRoot, nil)
+	if err != nil {
+		log.Errorf("failed to get mainnetExitRoot and blockNum by ger, err: %v", err)
 		return false
 	}
-	if numberOfBlocks >= s.cfg.WaitBlocksToUpdateGER {
-		if len(s.sequenceInProgress.Txs) == 0 {
-			log.Warn("TODO: update GER without closing batch as no txs have been added yet")
+
+	lastGer, err := s.state.GetLatestGlobalExitRoot(ctx, nil)
+	if err != nil {
+		log.Errorf("failed to get latest global exit root, err: %v", err)
+		return false
+	}
+
+	if lastGer.MainnetExitRoot != mainnetExitRoot {
```

Comment:
```suggestion
	if lastGer.GlobalExitRoot !=  s.sequenceInProgress.GlobalExitRoot
```
I think the logic should like this...